llarp/service/endpoint.cpp

```diff
@@ -86,18 +86,19 @@ namespace llarp
           ManualRebuild(1);
         return;
       }
-      introSet().I.clear();
+      std::vector< Introduction > intros;
       for(auto& intro : I)
       {
-        introSet().I.emplace_back(std::move(intro));
+        intros.emplace_back(std::move(intro));
       }
-      if(introSet().I.size() == 0)
+      if(intros.size() == 0)
       {
         LogWarn("not enough intros to publish introset for ", Name());
         if(ShouldBuildMore(now) || forceRebuild)
           ManualRebuild(1);
         return;
       }
+      introSet().I     = intros;
```

Comment:
I'm guessing the big difference here (between `introSet().I.clear()` earlier and `=` here) is that we leave the introset intact if we hit the `return` above. Assuming that's the motivation for this, what good does having a stale introset do?